catroid/src/main/java/org/catrobat/catroid/utils/StatusBarNotificationManager.java

```diff
@@ -331,4 +334,11 @@ private void closeNotificationBar() {
 			sendBroadcast(it);
 		}
 	}
+
+	public static int getUniqueNotificationId() {
+		if (uniqueNotificationIdCounter == null) {
+			uniqueNotificationIdCounter = new AtomicInteger(0);
+		}
```

Comment:
As I was calling `getUniqueNotificationId()` in `CustomFirebaseMessagingService `class then:-

When application was in background - Everything worked perfectly

When application was in foreground - The application was getting stopped unexpectedly (on receiving notification) without any error message but with a log [System.exit() is called with status 10]. After a while I realized that it was because of calling `getUniqueNotificationId()`. I did some changes and everything worked well.

 Seems like `uniqueNotificationIdCounter` wasn't getting initialized on directly calling static method  `getUniqueNotificationId()`.